backend/src/main/java/org/cryptomator/hub/validation/ValidUUID.java

```diff
@@ -0,0 +1,27 @@
+package org.cryptomator.hub.validation;
+
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.*;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Pattern(regexp = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}")
+@NotNull
+@Target({METHOD, FIELD, ANNOTATION_TYPE, TYPE_USE, PARAMETER})
+@Retention(RUNTIME)
+@Constraint(validatedBy = {})
+@Documented
+public @interface ValidUUID {
```

Comment:
I just noticed that we use IDs in our test data that are *not* UUIDs. Maybe we should drop this validation again? `ValidID` is sufficient to protect against malformed data.

Reply:
Done in https://github.com/cryptomator/hub/pull/154/commits/76ec596ab1308f5556c0bf73fbc9ded0ff876341